vespa/application.py

```diff
@@ -215,6 +215,8 @@ def get_application_status(self) -> Optional[Response]:
         :return:
         """
         endpoint = "{}/ApplicationStatus".format(self.end_point)
+        if self.key:
+            self.cert = (self.app.cert, self.app.key)
```

Comment:
This was handled by VespaSync before. I understand that you tried to simplify the get_application_status code in a previous PR but this might lead to duplication of code that is harder to maintain later.